bcda/api/requests.go

```diff
@@ -424,17 +422,61 @@ func (h *Handler) bulkRequest(w http.ResponseWriter, r *http.Request, reqType se
 	}
 }
 
-func (h *Handler) validateRequest(resourceTypes []string) error {
+func (h *Handler) getResourceTypes(parameters middleware.RequestParameters, cmsId string) []string {
+	resourceTypes := parameters.ResourceTypes
+
+	// If caller does not supply resource types, we default to all supported resource types for the specific ACO
+	if len(resourceTypes) == 0 {
+		if acoConfig, found := h.Svc.GetACOConfigForId(cmsId); found {
+			if contains(acoConfig.Data, "adjudicated") {
+				resourceTypes = append(resourceTypes, "Patient", "ExplanationOfBenefit", "Coverage")
+			}
+
+			if contains(acoConfig.Data, "preadjudicated") {
+				resourceTypes = append(resourceTypes, "Claim", "ClaimResponse")
+			}
+		}
+	}
+
+	return resourceTypes
+}
+
+func (h *Handler) validateRequest(resourceTypes []string, cmsId string) error {
 
 	for _, resourceType := range resourceTypes {
-		if _, ok := h.supportedResources[resourceType]; !ok {
+		resource, ok := h.supportedResources[resourceType]
+
+		if !ok {
 			return fmt.Errorf("invalid resource type %s. Supported types %s", resourceType, h.supportedResources)
 		}
+
+		if !h.authorizedResourceAccess(resource, cmsId) {
+			return fmt.Errorf("unauthorized resource type %s", resourceType)
+		}
 	}
 
 	return nil
 }
 
+func (h *Handler) authorizedResourceAccess(resource ResourceType, cmsId string) bool {
+	if cfg, ok := h.Svc.GetACOConfigForId(cmsId); ok {
+		return (resource.Adjudicated && contains(cfg.Data, "adjudicated")) ||
+			(resource.PreAdjudicated && contains(cfg.Data, "preadjudicated"))
+	}
+
+	return false
+}
+
+func contains(array []string, str string) bool {
```

Comment:
can we use `ContainsString` instead?  https://github.com/CMSgov/bcda-app/blob/master/bcda/utils/common.go#L63